src/aws-cpp-sdk-core/source/auth/signer/AWSAuthV4Signer.cpp

```diff
@@ -567,40 +560,38 @@ Aws::Utils::ByteBuffer AWSAuthV4Signer::ComputeHash(const Aws::String& secretKey
 {
     Aws::String signingKey(Aws::Auth::AWSAuthHelper::SIGNING_KEY);
     signingKey.append(secretKey);
-    auto hashResult = m_HMAC->Calculate(ByteBuffer((unsigned char*)simpleDate.c_str(), simpleDate.length()),
-            ByteBuffer((unsigned char*)signingKey.c_str(), signingKey.length()));
+    auto kDate = HashingUtils::CalculateSHA256HMAC(
+        ByteBuffer((unsigned char *) simpleDate.c_str(), simpleDate.length()),
```

Comment:
can we just add bytebuffer from string constructor?

Reply:
ByteBuffer is an Array, but yeah that would be a good convenience function let me add that